dataactvalidator/scripts/loadFile.py

```diff
@@ -144,6 +144,15 @@ def loadSF133(filename, fiscal_year, fiscal_period, force_load=False):
              "agency_identifier": {"pad_to_length": 3},
              "main_account_code": {"pad_to_length": 4},
              "sub_account_code": {"pad_to_length": 3},
+             # next 3 lines handle the TAS fields that shouldn't
+             # be padded but should still be empty spaces,
+             # at least temporarily, so that downstream formatting
+             # and cleaning functions work correctly. this works
+             # around the fact that sometimes the incoming data
+             # is a single space and sometimes it's blank.
+             "beginning_period_of_availa": {"pad_to_length": 0},
+             "ending_period_of_availabil": {"pad_to_length": 0},
+             "availability_type_code": {"pad_to_length": 0},
```

Comment:
That is a fair question and a long story.

**Short story**
We don't want to keep_null..the new sf-133 file had some nulls where it previously had spaces, which causes a downstream problem. This fix pads to 0 to change incoming nulls to spaces (everything is set back to null before the db load).

It would also be fair to ask, "why doesn't the SF-133 load have unit tests if it's so complicated?" If you want to hold the merge until said tests are written, I would not blame you.

**Long story**
The requirement to interpret missing rows and insert them as 0 values makes the SF-133 load different than the other domain loaders.

Here's the code at the crux of the problem: https://github.com/fedspendingtransparency/data-act-broker-backend/blob/hotfix-sf133-load/dataactvalidator/scripts/loadFile.py#L174

Lines 174-75 are a shortcut to fill in the 0 values. Line 174 uses `pivot_table` to explode the data out horizontally, creating one row for each tas/fiscal year/period, with columns for each SF-133 line. The `fill_value` parameter puts `0` into any SF-133 cell with a missing value for that tas/fiscal year/period.

Once we have the zeroes filled in, line 175 uses the `melt` function to re-arrange the data back to it's normal format of one row per tas/fiscal year/period.

It was the shortest and most efficient way I could think of to put in the missing zeroes. There's a catch, though--if any value that we pivot on in line 174 is null, that row is silently dropped by pandas. I lost several hours of my life tracking this down.

Which is a long way of saying...we do need the incoming TAS components to be blank values instead of NULLS to ensure that no lines are dropped in the pivot. Once we have the 0 value lines, we replace those empty strings with None before inserting to the db: https://github.com/fedspendingtransparency/data-act-broker-backend/blob/hotfix-sf133-load/dataactvalidator/scripts/loadFile.py#L196